src/coreclr/jit/importer.cpp

```diff
@@ -3942,6 +3942,52 @@ GenTree* Compiler::impImportStaticReadOnlyField(CORINFO_FIELD_HANDLE field, CORI
             uint8_t buffer[MaxStructSize] = {0};
             if (info.compCompHnd->getReadonlyStaticFieldValue(field, buffer, totalSize))
             {
+#ifdef FEATURE_SIMD
+                // First, let's check whether field is a SIMD vector and import it as GT_CNS_VEC
+                int simdWidth = getSIMDTypeSizeInBytes(fieldClsHnd);
+                if (simdWidth > 0)
+                {
+                    assert((totalSize <= 32) && (totalSize <= MaxStructSize));
+                    var_types simdType = getSIMDTypeForSize(simdWidth);
+
+                    bool hwAccelerated = false;
+                    switch (simdType)
+                    {
+#ifdef TARGET_ARM64
+                        case TYP_SIMD8:
+                        case TYP_SIMD12:
+                        case TYP_SIMD16:
+                            hwAccelerated = compExactlyDependsOn(InstructionSet_AdvSimd);
+                            break;
+#endif
+#ifdef TARGET_XARCH
+                        case TYP_SIMD12:
+                        case TYP_SIMD16:
+                            // Let's not complicate the test matrix with "for floating point SSE1 is enough"
+                            hwAccelerated = compExactlyDependsOn(InstructionSet_SSE2);
```

Comment:
This one and the `AdvSimd` case shouldn't be required, we can just return a constant `true`. `SSE2` and `AdvSimd` are "baseline" ISAs and `GT_CNS_VEC` doesn't depend on HWIntrinsics to perform the load/store (it directly emits the relevant instruction instead).